compiler/ballerina-lang/src/main/java/io/ballerina/projects/JarResolver.java

```diff
@@ -117,6 +117,19 @@ public Collection<JarLibrary> getJarFilePathsRequiredForExecution() {
         return jarFiles;
     }
 
+    public ClassLoader getClassLoaderWithRequiredJarFilesForExecutionWithoutMainJar() {
```

Comment:
Shall we move this to the shell module without keeping it in the Jar Resolver? 
JarResolver is expected to use by any API user and this method might confuse them since it's specific to the shell implementation.

Reply:
Updated in 0aad905fa9e08d086e402eea84e5cbe83d5f369f